src/components/MAPI/permissions/usePermissions.ts

```diff
@@ -0,0 +1,62 @@
+import { useAuthProvider } from 'Auth/MAPI/MapiAuthProvider';
+import { FlashMessage, messageTTL, messageType } from 'lib/flashMessage';
+import { useHttpClientFactory } from 'lib/hooks/useHttpClientFactory';
+import { extractErrorMessage } from 'MAPI/utils';
+import { GenericResponseError } from 'model/clients/GenericResponseError';
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { getLoggedInUser } from 'stores/main/selectors';
+import { LoggedInUserTypes } from 'stores/main/types';
+import { selectOrganizations } from 'stores/organization/selectors';
+import { IState } from 'stores/state';
+import useSWR from 'swr';
+
+import { IPermissionMap } from './types';
+import { fetchPermissions } from './utils';
+
+// eslint-disable-next-line no-magic-numbers
+const REFRESH_INTERVAL = 60 * 1000; // In ms.
+
+export function usePermissions() {
```

Comment:
This hook is the base for all permission hooks, and it handles fetching, storing and updating permissions